benchmarks/runner/benchmark_runner.py

```diff
@@ -0,0 +1,193 @@
+    out = _run_cmd(f"{python} -m pip show fractal_tasks_mock")
+    location = next(
+        line for line in out.split("\n") if line.startswith("Location:")
+    )
+    location = location.replace("Location: ", "")
+    src_dir = Path(location) / "fractal_tasks_mock/"
+
+    with (src_dir / "__FRACTAL_MANIFEST__.json").open("r") as f:
+        manifest = json.load(f)
+
+    task_dict = {}
+    for ind, task in enumerate(manifest["task_list"]):
+        args = {}
+        if task.get("executable_non_parallel"):
+            args[
+                "command_non_parallel"
+            ] = f"{python} {src_dir / task['executable_non_parallel']}"
+            args["meta_non_paralell"] = task.get("meta_non_paralell")
+        if task.get("executable_parallel"):
+            args[
+                "command_parallel"
+            ] = f"{python} {src_dir / task['executable_parallel']}"
+            args["meta_paralell"] = task.get("meta_paralell")
+
+        t = TaskV2Mock(
+            id=ind,
+            name=task["name"],
+            source=task["name"].replace(" ", "_"),
+            input_types=task.get("input_types", {}),
+            output_types=task.get("output_types", {}),
+            **args,
+        )
+        task_dict[t.name] = t
+
+    return task_dict
+
+
+venv_dir = tempfile.mkdtemp()
+fractal_tasks_mock_venv = mock_venv(venv_dir)
+
+
+def benchmark(N: int, tmp_path: str):
+
+    WORKING_DIR = Path(f"{tmp_path}/job_dir")
+    ZARR_DIR = (WORKING_DIR / "zarr").as_posix().rstrip("/")
+
+    execute_tasks_v2(
+        wf_task_list=[
+            WorkflowTaskV2Mock(
+                task=fractal_tasks_mock_venv["create_ome_zarr_compound"],
+                args_non_parallel=dict(image_dir=ZARR_DIR, num_images=N),
+                id=0,
+                order=0,
+            ),
+            WorkflowTaskV2Mock(
+                task=fractal_tasks_mock_venv["illumination_correction"],
+                args_parallel=dict(overwrite_input=True),
+                id=1,
+                order=1,
+            ),
+            WorkflowTaskV2Mock(
+                task=fractal_tasks_mock_venv["MIP_compound"],
+                id=2,
+                order=2,
+            ),
+            WorkflowTaskV2Mock(
+                task=fractal_tasks_mock_venv["cellpose_segmentation"],
+                id=3,
+                order=3,
+            ),
+        ],
+        dataset=DatasetV2Mock(name="dataset", zarr_dir=ZARR_DIR),
+        workflow_dir=WORKING_DIR,
+        workflow_dir_user=WORKING_DIR,
+        executor=FractalThreadPoolExecutor(),
+    )
+
+
+if __name__ == "__main__":
+    results = []
+
+    for N in range(100, 1001, 100):
+
+        tmp_path = tempfile.mkdtemp()
+        cProfile.run(f"benchmark({N}, '{tmp_path}')", "profile_results")
+        stats = pstats.Stats("profile_results")
+        stats.sort_stats("tottime")
+
+        thread_time = stats.stats[
+            "~", 0, "<method 'acquire' of '_thread.lock' objects>"
```

Comment:
Let's review together that this is the right approach (with an artificial example).

Reply:
https://github.com/fractal-analytics-platform/fractal-server/issues/1420#issuecomment-2074920560